docs/using-the-jdbc-driver/UsingTheJdbcDriver.md

```diff
@@ -100,6 +100,25 @@ DriverConfigurationProfiles.addOrReplaceProfile(
         CustomConnectionPluginFactory.class));
 ```
 
+### Executing Custom Code When Initializing a Connection
+Users may need to define a specific configuration for a connection that has just been opened by the driver before an application can use it. Not all target drivers provide this functionality, but some allow specifying a configuration parameter with SQL statements that are executed when a connection is established. Some cases may also require that additional conditions are checked in order to identify what initialization configuration is required for a particular connection.
+
+AWS JDBC Driver allows to specify a special function that can initialize a connection. It can be done with `ConnectionProviderManager.setConnectionInitFunc` method. `resetConnectionInitFunc` method is also available. 
+
+The initialization function is called for all connections, including pre-opened connections provided by internal connection pools (see [Using Read Write Splitting Plugin and Internal Connection Pooling](./using-plugins/UsingTheReadWriteSplittingPlugin.md#internal-connection-pooling)). This helps user applications clean up connection sessions that have been altered by previous operation, as returning a connection to a pool will reset the state and retrieving it will call the initialization function again.
```

Comment:
```suggestion
The initialization function is called for all connections, including connections opened by the internal connection pools (see [Using Read Write Splitting Plugin and Internal Connection Pooling](./using-plugins/UsingTheReadWriteSplittingPlugin.md#internal-connection-pooling)). This helps user applications clean up connection sessions that have been altered by previous operations, as returning a connection to a pool will reset the state and retrieving it will call the initialization function again.
```

Reply:
We're saying that returning a connection to the pool resets its session state, is this accurate though? If I understand this PR correctly, the only extra logic is performed when retrieving a connection, not when returning it, and whether the session state is fully reset or not depends on what the user defines in their ConnectionInitFunc object.

Reply:
Hikari CP does reset some connection/session settings while returning connection back to the pool. Hikari tracks just a limited number of session settings like read-only, transaction isolation, etc. It doesn't track temporary tables, language settings and other that can be essential for next user transactions.  This function is about setting them while getting connection from the pool, or while opening a brand new connection.